repository/src/main/java/com/epam/esm/repository/impl/UserRepositoryImpl.java

```diff
@@ -0,0 +1,54 @@
+package com.epam.esm.repository.impl;
+
+import com.epam.esm.entity.User;
+import com.epam.esm.repository.PaginationContext;
+import com.epam.esm.repository.repositoryinterfaces.UserRepository;
+import org.springframework.beans.factory.annotation.Autowired;
+import org.springframework.stereotype.Repository;
+import org.springframework.transaction.annotation.Transactional;
+
+import javax.persistence.EntityManager;
+import javax.persistence.PersistenceContext;
+import java.util.List;
+import java.util.Optional;
+
+import static com.epam.esm.repository.builder.UserQueries.FIND_ALL_USERS;
+
+@Repository
+public class UserRepositoryImpl implements UserRepository {
+
+    @PersistenceContext
+    private EntityManager entityManager;
+    private final PaginationContext paginationContext;
+
+    @Autowired
+    public UserRepositoryImpl(EntityManager entityManager, PaginationContext paginationContext) {
+        this.paginationContext = paginationContext;
+        this.entityManager = entityManager;
+    }
+
+    @Override
+    public List<User> findAll() {
+        return entityManager.createQuery(FIND_ALL_USERS, User.class)
```

Comment:
Can we move `findAll` method to abstract repository? The only things changed are class and query

Reply:
You could also extract all another methods of repository

Reply:
fixed, but now sure